structuralcodes/materials/concrete/_concreteEC2_2023.py

```diff
@@ -0,0 +1,194 @@
+    _gamma_C = t.Optional[float] = None  # noqa: N815
+
+    # Computed attributes
+    _fcm = t.Optional[float] = None
+    _fctm = t.Optional[float] = None
+    _Ecm = t.Optional[float] = None
+
+    # TODO: do we agree with the optional keyword arguments here?
+    def __init__(
+        self,
+        fck: float,
+        name: t.Optional[str] = None,
+        density: float = 2400.0,
+        kE: float = 9500,
+        concrete_dev_class: str = 'CS',
+        gamma_C: float = 1.5,
+        existing: bool = False,
+    ):
+        """Initializes a new instance of Concrete for MC 2010.
+
+        Args:
+            fck (float): Characteristic strength in MPa if concrete is not
+                existing.
+
+        Keyword Args:
+            name (str): A descriptive name for concrete
+            density (float): Density of material in kg/m3 (default: 2400)
+            kE (float): coefficient relating aggregates.
+            concrete_dev_class (str, optional): default is CS
+            gamma_C (float, optioanl): partial factor of concrete
+                (default is 1.5)
+            existing (bool, optional): The material is of an existing structure
+                (default: False)
+        """
+        if name is None:
+            name = f'C{round(fck):d}'
+        super().__init__(
+            fck=fck, name=name, density=density, existing=existing
+        )
+        self._kE = kE
+        self._concrete_dev_class = concrete_dev_class
+        self._gamma_C = gamma_C
+
+        # TODO: implement this method
+        def _reset_attributes(self):
+            # We only need to reset computed properties
+            self._fcm = None
+            self._fctm = None
+            self._Ecm = None
+
+        @property
+        def fcm(self) -> float:
+            """Returns the mean strength of concrete.
+
+            Returns:
+                float: The mean compressive strength in MPa.
+            """
+            if self._fcm is not None:
+                return self._fcm
+            self._fcm = ec2_2023.fcm(self.fck)
+            return self._fcm
+
+        @fcm.setter
+        def fcm(self, value: float):
+            """Sets a user defined value for the mean strength of concrete.
+
+            Args:
+                value (float): the value of the mean strength of
+                    concrete in MPa.
+
+            Raises:
+                ValueError: if value is less or equal than the value of fck
+            """
+            if abs(value) <= self._fck:
+                raise ValueError(
+                    (
+                        'Mean compressive strength cannot be lower than',
+                        'characteristic strength.\n',
+                        'Current characteristing strength: ',
+                        f'fck = {self._fck}.',
+                        f'Current value: value = {value}',
+                    )
+                )
+            self._reset_attributes()
```

Comment:
We should not reset the attributes in the setters. If a user first set e.g. `fcm`, and furthermore `fctm`, the value that was set to `fcm` will be replaced with `None`. This comment also applies to the setters of `fctm` and `Ecm`.

Reply:
I agree. reset is called by parent class `Concete` when `fck` is set. Indeed when `fck` is set as a new value is like having a "totally different" concrete, therefore the reset.

Reply:
Done!